pkg/util/math/rate.go

```diff
@@ -0,0 +1,59 @@
+package math
+
+import (
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+)
+
+// EwmaRate tracks an exponentially weighted moving average of a per-second rate.
+type EwmaRate struct {
+	newEvents atomic.Int64
+
+	alpha    float64
+	interval time.Duration
+
+	mutex    sync.RWMutex
+	lastRate float64
+	init     bool
+}
+
+func NewEWMARate(alpha float64, interval time.Duration) *EwmaRate {
```

Comment:
Is this used somewhere ? Not sure it is.

Reply:
Yea you are right. Removed it.